Sourcery/Parsing/FileParser.swift

```diff
@@ -234,8 +234,25 @@ extension FileParser {
 
     internal func parseVariable(_ source: [String: SourceKitRepresentable], isStatic: Bool = false) -> Variable? {
         guard let (name, _, accesibility) = parseTypeRequirements(source),
-            accesibility != .private && accesibility != .fileprivate,
-            let type = source[SwiftDocKey.typeName.rawValue] as? String else { return nil }
+            accesibility != .private && accesibility != .fileprivate else { return nil }
+
+        var maybeType: String? = source[SwiftDocKey.typeName.rawValue] as? String
```

Comment:
Can we do the same not by manually parsing string, but using structure output? In case of `var some = String()` it will give us

```
{
      "key.kind" : "source.lang.swift.expr.call",
      "key.offset" : 12,
      "key.nameoffset" : 12,
      "key.namelength" : 6,
      "key.bodyoffset" : 19,
      "key.bodylength" : 0,
      "key.length" : 8,
      "key.name" : "String"
    }
```
The same will be for arrays or dictionaries when we use `[String]()` or `[String: Int]()`. For `Foo(some: Int)`, it will report just `Foo` in `key.name`, in case of `Foo.init(...)`, it will report `Foo.init`, then we just trim `init` part. Btw there also could be any static method calls, like `Foo.makeBar()`, then we would need to get return type of that method...